packages/fastify-renderer/src/node/RenderBus.tsx

```diff
@@ -1,15 +1,51 @@
+import { Readable } from 'stream'
+
+export interface Stack {
+  content: string[]
+  hasEnded: boolean
+  contentStreamed: boolean
+  stream: Readable
+}
+
 /** Holds groups of content during a render that eventually get pushed into the template. */
 export class RenderBus {
-  stacks: Record<string, string[]> = {}
+  stacks: Record<string, Stack> = {}
   included = new Set<string>()
 
-  push(key: string, content: string) {
-    this.stacks[key] ??= []
-    this.stacks[key].push(content)
+  private createStack(key) {
+    const stack: Stack = (this.stacks[key] = {
+      content: [],
+      hasEnded: false,
+      contentStreamed: false,
+      stream: new Readable(),
+    })
+
+    stack.stream._read = function () {
+      if (stack.hasEnded && stack.contentStreamed) {
+        this.push(null)
+      } else {
+        this.push(stack.content.join('\n'))
+        stack.contentStreamed = true
+      }
+    }
+
+    return stack
+  }
+
+  push(key: string, content: string | null) {
+    if (!this.stacks[key]) this.createStack(key)
+    if (this.stacks[key].hasEnded) return
```

Comment:
Maybe throw an error here since no content should be pushed to the stack after pushing `null`?

Reply:
Ya I think thats a good idea, that'd be a developer's mistake or ours